### ISSUE TITLE
Table width as percentage is miscalculated

### ISSUE THREAD

**Issue by jbjorge** [bpampuch/pdfmake#401]
If tables are created with widths specified as percentage, the columns do not align.

![screenshot from 2015-10-06 15 08 45](https://cloud.githubusercontent.com/assets/1926428/10309341/4166a9b0-6c3c-11e5-94a2-bec276218f99.png)

``` javascript
var dd = {
    content: [
        {
            table: {
                widths: ['25%', '75%'],
                body: [['25%', '75%']]
            }
        },
        {
            table: {
                widths: ['25%', '25%', '25%', '25%'],
                body: [['25%', '25%', '25%', '25%']]
            }
        }
    ]
}
```

I've looked at the source, but find it hard to pinpoint exactly where the miscalculation occurs.
Are there any workarounds for this? (or hints so I can make a pull request)

**Comment by AgDude** [comment]
I strongly suspect this is related to pdfmake/pdfmake#30. I have not confirmed, but I suspect that the percents are calculated before accounting for padding and borders.

**Comment by jafar-rezaei** [comment]
any updates ?

**Comment by jbjorge** [comment]
@jafar-rezaei To mitigate the issue when I first stumbled upon it I switched away from using percent when defining widths. Since the pdf is scaled after a physical format, the sizes for the page is known from [standardPageSizes.js](https://github.com/bpampuch/pdfmake/blob/e3988af8baccd8a2a4eea0ffcf6a68506b937900/src/standardPageSizes.js).
The size of the table padding and vertical lines are known through
```javascript
{
  table: { ... },
  layout: {
    paddingLeft: function(i, node) { return 4; },
    paddingRight: function(i, node) { return 4; },
    hLineWidth: function (i, node) { return 1; },
    vLineWidth: function (i, node) { return 1; },
  }
}
```
and the size of the page margins are known through
```javascript
const docDefinition = {
    pageSize: [595.28, 841.89],
    pageMargins: [ 10, 10 ]
}
```
Things in pdfmake may have been changed since I implemented it in our codebase, but if I remember correctly i had to account for those things when calculating 25% width out of a page that is e.g. A4.
